common/src/main/java/com/wynntils/models/items/encoding/data/NameData.java

```diff
@@ -6,9 +6,22 @@
 
 import com.wynntils.models.items.encoding.type.ItemData;
 import com.wynntils.models.items.properties.IdentifiableItemProperty;
+import java.util.regex.Pattern;
 
 public record NameData(String name) implements ItemData {
+    private static final int MAX_NAME_LENGTH = 48;
+    private static final Pattern SANITIZE_PATTERN = Pattern.compile("[^a-zA-Z0-9'\\s]");
+
     public static NameData from(IdentifiableItemProperty property) {
-        return new NameData(property.getName());
+        return NameData.from(property.getName());
+    }
+
+    public static NameData from(String name) {
+        name = SANITIZE_PATTERN.matcher(name).replaceAll("");
+        name = name.trim();
+        name = name.replaceAll("\\s\\s", " ");
```

Comment:
Shouldn't this be
```suggestion
        name = name.replaceAll("\\s+", " ");
```

Otherwise e.g. "\t\t\t" would be " \t", right?

Reply:
@kristofbolyai Do you agree with this?